samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -524,10 +468,125 @@ public void onStreamProcessorLaunchFailure(SamzaResource resource, Throwable t)
    */
   @Override
   public void onError(Throwable e) {
-    log.error("Exception occurred in callbacks in the Cluster Resource Manager", e);
+    LOG.error("Exception occurred in callbacks in the Cluster Resource Manager", e);
     exceptionOccurred = e;
   }
 
+  /**
+   * Called within {@link #onResourceCompleted(SamzaResourceStatus)} for unknown exit statuses. Usually these type of
+   * exit statuses are due to application errors causing the container resource to fail or for other unknown reasons.
```

Comment:
Nitpick:
These exit statuses correspond to container completion other than container run-to-completion, abort or preemption, or disk failure (e.g., detected by YARN's NM healthchecks).